src/ucp/core/ucp_context.h

```diff
@@ -320,6 +320,10 @@ typedef struct ucp_context {
        ucp_mem_map() will register memory for all those domains. */
     ucp_md_map_t                  reg_md_map[UCS_MEMORY_TYPE_LAST];
 
+    /* Map of MDs that provide nonblocking registration for given memory type,
+       ucp_mem_map() will register memory for all those domains. */
+    ucp_md_map_t                  reg_nonblock_md_map[UCS_MEMORY_TYPE_LAST];
```

Comment:
The description for the next field `reg_block_md_map` is confusing.
It is said that `reg_md_map` contains only MDs supporting non-blocking registration. Which means that the new map is redundant.

Reply:
maybe add something like: `.. when nonblocking is requested, registration will only be done for MD part of the Map`